src/main/java/com/iexec/resultproxy/ipfs/IpfsService.java

```diff
@@ -21,17 +40,24 @@ public class IpfsService implements SmartLifecycle {
     private IPFS ipfs;
     private final String multiAddress;
 
-    public IpfsService(IpfsConfig ipfsConfig) {
-        String ipfsHost = ipfsConfig.getHost();
-        String ipfsNodeIp = NetworkUtils.isIPAddress(ipfsHost) ? ipfsHost : NetworkUtils.convertHostToIp(ipfsHost);
-        this.multiAddress = "/ip4/" + ipfsNodeIp + "/tcp/" + ipfsConfig.getPort();
+    public IpfsService(final IpfsConfig ipfsConfig) {
+        try {
+            final URL ipfsUrl = new URL(ipfsConfig.getUrl());
+            final String ipfsHost = ipfsUrl.getHost();
+            final int port = ipfsUrl.getPort() != -1 ? ipfsUrl.getPort() : ipfsUrl.getDefaultPort();;
+            final String ipfsNodeIp = convertHostToIp(ipfsHost);
+            this.multiAddress = "/ip4/" + ipfsNodeIp + "/tcp/" + port;
+        } catch (MalformedURLException e) {
```

Comment:
We want to catch all kind of exceptions here
```suggestion
        } catch (Exception e) {
```

Reply:
Resolved in [eed8f2d](https://github.com/iExecBlockchainComputing/iexec-result-proxy/pull/155/commits/eed8f2dd6ba49099966d5c093bc994b331e707c3)